### PR TITLE
ARROW-1270: [Python] Build scripts for creating macOS Python wheels in Travis CI

### PR DESCRIPTION
**Pull request by wesm**
Also includes ARROW-1272 since they're both on Travis CI

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,24 +15,80 @@
...
+after_success:
+- echo "After success"
+- pwd
+
+deploy:
+  provider: bintray
+  file: "descriptor.json"
+  user: xhochy
+  key:
+    secure: Fb4R4YAhTCXGtavVB4CtHzWkQeJiny4rAuyJbScgQkaqaT7SG+rywA8pIOHPSSPf8yRiz25Z/+8aVEFvWqcHLQtFf8peTiJMmqYoMCQf4WjV1wldDwysr+NinGs0jbEAqAWJ127mNLELdn1wIduBXupgTG8MKFH+aR2O8Rdi2S5Od6Xk1TPLlEsvSlsN4IHGKN7qHsCeiMbzPM/R3lWyaNa92m1qhd2K8ypUtBPKgqemp0ZKrNHTDmAs6PmCNryPLA71jcwA9mPwVRfV+H6/4xNKpnQkFC2TfMok7JeMvru4wKhgWtmU0J0t4LYHqNcNQyhqlQD60VEwNs987bTPTKRPbORdgyP1v/ArthQEDD61o7V8jazoXudL8SwQj/mFPzF+o1t5D06300MCesSijuo0UgvvbtyZHN8SzudxUCZSrB+mSXeJ5GR2viSY3ndxqwPH2wMi/ZzO93Qxr2b8G7gTwDrX9ffhRkSL/NchdxRuUwOf5ejp8Yy2qVyjFMEMZWNemNhsFg4qDCAxv0pnAk/0N0kPE0wXrVbQW3kK/JywtV03r8GNiemqfm23d4KTwPH21FU4FpCbEpT4S7yUzBju4xpY9JyPBnvi8ZeQoqG5FUv+CcMUiCf2nPdil9RAs1maK2D6WPnSI02AicIkY0bv8/gXTsSgXzncOsxi/+E=
```
Comment (xhochy) [inline .travis.yml]: This will probably not work as it's encrypted for a different repo

Comment (wesm) [reply .travis.yml]: I'll generate a new key to deploy these (including Windows wheels) to a central bintray. Can we get an Apache bintray?

Comment (xhochy) [reply .travis.yml]: We already have an  Apache bintray: https://bintray.com/apache/arrow

Comment (wesm) [reply .travis.yml]: How do I get permissions to administer that account?

Comment (xhochy) [reply .travis.yml]: Create a personal bintray account and then make an INFRA JIRA so that INFRA can give you the access to it.

Comment (wesm) [reply .travis.yml]: Thanks. I opened INFRA-14783

**Comment by wesm** [comment]
The services will not deploy from pull requests, so I'm going to "test it in production" to https://bintray.com/wesm/apache-arrow-test

**Pull request merged**